src/cracker.c

```diff
@@ -996,24 +989,11 @@ static int crk_salt_loop(void)
 	if (!salt || crk_db->salt_count < 2)
 		status.resume_salt_md5 = NULL;
 
-	if (done >= 0) {
-#if !HAVE_OPENCL
-		/* Assumes we'll never overrun 32-bit in one crypt */
-		add32to64(&status.cands, crk_key_index *
-		          mask_int_cand.num_int_cand);
-#else
-		/* Safe for > 4G crypts per call */
-		int64 totcand;
-		mul32by32(&totcand, crk_key_index, mask_int_cand.num_int_cand);
-		add64to64(&status.cands, &totcand);
-#endif
-	}
+	if (done >= 0)
+		status.cands += crk_key_index * mask_int_cand.num_int_cand;
```

Comment:
I think this is a bug: we need explicit cast to `uint64_t` before the multiplication.